### ISSUE TITLE
RecordFormat[T].from(...) not picking up default values of schema

### ISSUE THREAD

**Issue by pjan** [sksamuel/avro4s#110]
How to reproduce:

```
case class P1(name: String, age: Int = 18)
case class P2(name: String)

val f1 = RecordFormat[P1]
val f2 = RecordFormat[P2]

f1.from(f2.to(P2("test")))
```

This blows up in a nullpointer exception during runtime. Expected behaviour would be that the default value of the schema gets picked up, and the code doesn't blow up. Or am I missing something?

**Comment by sksamuel** [comment]
Try making the formats implicit.

**Comment by pjan** [comment]
Not sure how this would solve anything, since nothing in the code example requires implicit `formats`.

Anyway, tried making them implicit, and it doesn't change anything, unfortunately...

**Comment by sksamuel** [comment]
Never mind, misunderstood.

**Comment by pjan** [comment]
It does work when I use output & inputstreams, so I am probably mis-using the api in the example above. Can close if that would be the case/suspicion.

**Comment by Bhashit** [comment]
Not to push or anything, but is anyone working on this? If not, would you mind if I took this up? I am getting the same issue.

**Comment by sksamuel** [comment]
@Bhashit please feel free to tackle it

**Comment by sksamuel** [comment]
This is fixed in 2.0.0-M3 onwards.

See test using your code added here:
https://github.com/sksamuel/avro4s/blob/master/avro4s-core/src/test/scala/com/sksamuel/avro4s/github/GithubIssue110.scala
